src/lang/modifyAst/addFillet.test.ts

```diff
@@ -87,7 +87,8 @@ const runGetPathToExtrudeForSegmentSelectionTest = async (
       return expedtedExtrudeNodeResult
     }
     const expectedExtrudeNode = expedtedExtrudeNodeResult.node
-    const init = expectedExtrudeNode.declarations[0].init
+    const init =
+      expectedExtrudeNode.declarations?.[0].init || expectedExtrudeNode.init
```

Comment:
This is really weird. If I do `yarn test` without this change, I get failures:

```
 FAIL  src/lang/modifyAst/addFillet.test.ts > Testing getPathToExtrudeForSegmentSelection > should return the correct paths for a valid selection and extrusion
TypeError: Cannot read properties of undefined (reading 'type')
 ❯ getExpectedExtrudeExpression src/lang/modifyAst/addFillet.test.ts:92:14
     90|     const init =
     91|       expectedExtrudeNode.declarations?.[0].init 
     92|     if (init.type !== 'CallExpression' && init.type !== 'PipeExpression') {
       |              ^
     93|       return new Error(
     94|         'Expected extrude expression is not a CallExpression or PipeExpression'
 ❯ runGetPathToExtrudeForSegmentSelectionTest src/lang/modifyAst/addFillet.test.ts:138:39
 ❯ src/lang/modifyAst/addFillet.test.ts:161:5
```

But if I add this line, the tests pass. On the other hand, `yarn tsc` fails with this change. Any idea what's going on?

Reply:
I believe it's related to #3207, which is a PR that I worked on to try to fix this issue. The return type of the `node: T` of `getNodeFromPath<T>()` is a lie. The node is asserted to be `T`, but it's not actually guaranteed to be that type at all.

In this case, it seems like it's actually returning a `VariableDeclarator`, not a `VariableDeclaration`. I hope you don't mind that I pushed a commit showing that this change both passes the tests and `tsc`. It was easier to commit it than explain exactly what you needed to do.

Reply:
I owe you a beer etc at kittycamp!!!